js/ssai_pulse.js

```diff
@@ -0,0 +1,586 @@
+     *                        following fields:
+     *                 code : The amc.AD_CANCEL_CODE for the cancellation
+     */
+    this.cancelAd = function(ad, params) {
+    };
+
+    /**
+     * Called by Ad Manager Controller.  The ad manager should pause the ad passed to the function as a
+     * parameter.  If the given ad is not currently playing, no action is required.
+     * @method SsaiPulse#pauseAd
+     * @public
+     * @param {object} ad The ad object to pause
+     */
+    this.pauseAd = function(ad) {
+    };
+
+    /**
+     * Called by Ad Manager Controller.  The ad manager should resume the ad passed to the function as a
+     * parameter.  If the given ad is not currently loaded or not paused, no action is required.
+     * @method SsaiPulse#resumeAd
+     * @public
+     * @param {object} ad The ad object to resume
+     */
+    this.resumeAd = function(ad) {
+    };
+
+    /**
+     * <i>Optional.</i><br/>
+     * When the Ad Manager Controller needs to hide the overlay it will call this function.
+     * NOTE: This function should only be used by the ad manager if the cancelOverlay function is not being used.
+     * NOTE 2: Only implement this function if you plan to hide and reshow the overlay. Otherwise delete it or leave it commented.
+     * @method SsaiPulse#hideOverlay
+     * @public
+     * @param {object} currentAd The overlay ad object to be stored so when it is shown again, we can update the AMC
+     */
+    //this.hideOverlay = function(currentAd) {
+    //};
+
+    /**
+     * <i>Optional.</i><br/>
+     * When the Ad Manager Controller needs to cancel the overlay it will call this function.
+     * NOTE: This function should only be used by the ad manager if the hideOverlay function is not being used.
+     * NOTE 2: Only implement this function if you plan to cancel and not reshow the overlay. Otherwise leave it commented or delete it.
+     * @method SsaiPulse#cancelOverlay
+     * @public
+     * @param {object} currentAd The overlay ad object that the ad manager needs to know is going to be cancelled and removed
+     */
+    //this.cancelOverlay = function(currentAd) {
+    //};
+
+    /**
+     * This function gets called by the ad Manager Controller when an ad has completed playing. If the main video is
+     * finished playing and there was an overlay displayed before the post-roll then it needs to be removed. If the main
+     * video hasn't finished playing and there was an overlay displayed before the ad video played, then it will show
+     * the overlay again.
+     * @method SsaiPulse#showOverlay
+     * @public
+     */
+    this.showOverlay = function() {
+    };
+
+    /**
+     * <i>Optional.</i><br/>
+     * Called when player clicks on the tap frame, if tap frame is disabled, then this function will not be
+     * called
+     * @method SsaiPulse#playerClicked
+     * @public
+    */
+    this.playerClicked = function(amcAd, showPage) {
+      window.open(amcAd.ad.clickthrough);
```

Comment:
i know this probably came from the template but need to check that amcAd and amcAd.ad exist